composite-base/src/Composite/Record.hs

```diff
@@ -82,6 +83,15 @@ instance Monad ((:->) s) where
   Val a >>= k = k a
   {-# INLINE (>>=) #-}
 
+instance NFData a => NFData (s :-> a) where
+  rnf (Val x) = rnf x
+
+instance NFData (Record '[]) where
+  rnf RNil = ()
+
+instance (NFData a, NFData (Record xs), x ~ (s :-> a)) => NFData (Record (x : xs)) where
```

Comment:
this should probably be `instance (NFData (f a), NFData (Rec f rs)) => NFData (Rec f (x ': xs))` such that it works for all record types and field types as long as they themselves can be reduced to normal form. also I'm surprised the compiler didn't complain about unticked `:` in the type... are you compiling with the Makefile so that warnings are treated as errors?

Reply:
I haven't been, there are other errors besides this if I hit make though. Will push.